Makefile

```diff
@@ -56,7 +56,12 @@ endif
 
 go_clientgen := $(shell which client-gen)
 ifeq ($(go_clientgen),)
-go_clientgen := cd hack/ci-deps && go install k8s.io/code-generator/cmd/client-gen@v0.22.2 && cd ../.. && test -x "${GOPATH}/bin/client-gen"
+go_clientgen := cd hack/ci-deps && go install k8s.io/code-generator/cmd/client-gen@v0.22.2 && cd ../.. && "${GOPATH}/bin/client-gen"
+endif
+
+go_controllergen := $(shell which controller-gen)
+ifeq ($(go_controllergen),)
+go_controllergen := cd hack/ci-deps && go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.8.0 && cd ../.. && "${GOPATH}/bin/controller-gen"
```

Comment:
`&& "${GOPATH}/bin/controller-gen"` should probably be echo or something?

Reply:
I guess no, it should just call this binary, otherwise it doesn't work.

They way it is used in further code is ```
$(go_controlergen) <controller-gen-args>
```

Reply:
Hmm right, the whole thing is a string.

This is how it's done in k0sctl:

```makefile
golint := $(shell which golangci-lint)
ifeq ($(golint),)
golint := $(shell go env GOPATH)/bin/golangci-lint
endif

$(golint):
	go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```